wremnants/CardTool.py

```diff
@@ -586,30 +588,36 @@ def fillCardWithSyst(self, syst):
 
         # Deduplicate while keeping order
         systNames = list(dict.fromkeys(names))
-        if systInfo["decorrCharge"]:
-            systNames = list([f"{x}_q{i}" for x in systNames for i in range(2)])
 
         systnamesPruned = [s for s in systNames if not self.isExcludedNuisance(s)]
         systNames = systnamesPruned[:]
-        for systname in systNames:
-            if type(scale) == dict:
-                for reg in scale.keys():
-                    if re.match(reg, systname):
-                        thiscale = str(scale[reg])
-                        include = [(thiscale if x in procs else "-").ljust(self.procColumnsSpacing) for x in nondata]
-                        break # exit this inner loop when match is found, to save time
-            shape = "shape" if not systInfo["noConstraint"] else "shapeNoConstraint"
-            for chan in self.channels:
+        for chan in self.channels:
+            systNamesChan = []
+            if systInfo["decorrCharge"]:
+                for x in systNames:
+                    if self.chargeIdDict[chan]['id'] in x or (self.chargeIdDict[chan]['badId'] is not None and self.chargeIdDict[chan]["badId"] in x):
+                        logger.warning(f"decorrCharge action requested for syst {x} whose name already contains channel keywords. Please check")
+                    systNamesChan.append(f"{x}_{self.chargeIdDict[chan]['id']}")
+                    
+            else:
+                systNamesChan = systNames[:]
+            for systname in systNamesChan:
+                if type(scale) == dict:
```

Comment:
Where is the scaling via a dictionary used? It scales different groups by different amounts?

Reply:
It was originally introduced for the efficiencies to split effStat groups, while also keeping the global one. In general it might be needed anytime we have a boost histograms grouping many variations but for which we may want to check the impacts in bunches (I don't know, could be the QCD scales for different helicity fractions)

Reply:
Ok, this is fair. Is it used somewhere, though? I feel like this loop over channels and systs is too complicated. Do we really need regex matching for it? You really expect to scale separately by channel? I would have thought it could just be scale = scale if type(scale) != dict or proc not in scale else scale[proc]